crowbar_framework/app/models/node_object.rb

```diff
@@ -1321,12 +1321,57 @@ def boot_device!(device)
     boot_device(device) and save
   end
 
+  # TODO: Remove duplicate code and use a gem/git submodule/whatever...
+  # see barclamp-deployer/chef/cookbooks/barclamp/libraries/barclamp_library.rb
+  def unique_name_already_claimed_by(device)
+    claimed_disks = @node[:crowbar_wall][:claimed_disks].to_hash
+    claimed_disks.each do | claimed_name, v |
+      if self.link_to_device?(device, claimed_name)
+        return claimed_name
+      end
+    end
+    nil
+  end
+
+  # TODO: Remove duplicate code and use a gem/git submodule/whatever...
+  # see barclamp-deployer/chef/cookbooks/barclamp/libraries/barclamp_library.rb
+  #
+  # is the given linkname a link to the given device? In attributes,
+  # that means that the given linkname can be found in the array
+  # @node[:block_device][device][by-{id,path,uuid}]
+  def link_to_device?(device, linkname)
+    # device is i.e. "sda", "vda", ...
+    # linkname is i.e. "/dev/disk/by-path/pci-0000:00:04.0-virtio-pci-virtio1"
+    meta = @node[:block_device][device]
+    if meta and meta[:disks]
+      lookup_and_name = linkname.gsub(/^\/dev\/disk\//, '').split(File::SEPARATOR)
```

Comment:
Does `linkname` value depend on `meta[:disks]` somehow (possibly being nil because it is constructed from it)? If not, it might be clearer to move this before the horrible if nest, to make it at least tiny bit more readable.